public/app/plugins/datasource/grafana-azure-monitor-datasource/components/SubscriptionField.tsx

```diff
@@ -0,0 +1,84 @@
+import { SelectableValue } from '@grafana/data';
+import { Select } from '@grafana/ui';
+import React, { useCallback, useEffect, useState } from 'react';
+import { AzureMonitorQuery, AzureQueryType } from '../types';
+import { toOption, findOption } from './common';
+import { Field } from './Field';
+import { AzureQueryEditorFieldProps, Option } from '../types';
+
+interface SubscriptionFieldProps extends AzureQueryEditorFieldProps {
+  onQueryChange: (newQuery: AzureMonitorQuery) => void;
+}
+
+const SubscriptionField: React.FC<SubscriptionFieldProps> = ({ datasource, onQueryChange, query }) => {
+  const [subscriptions, setSubscriptions] = useState<Option[]>([]);
+
+  useEffect(() => {
+    if (!datasource.azureMonitorDatasource.isConfigured()) {
+      return;
+    }
+
+    console.log('SubscriptionField effect ');
+
+    datasource.azureMonitorDatasource.getSubscriptions().then((results) => {
+      console.log('getSubscriptions results', results);
+      const newSubscriptions = results.map(toOption);
+      setSubscriptions(newSubscriptions);
+
+      // TODO: how much of this is needed?
+      // let newSubscription: string;
+
+      // if (!query.subscription && query.queryType === 'Azure Monitor') {
```

Comment:
Good question..I don't know!

Reply:
at least thanks for the reminder to look into this more!